src/reanimated2/animation/spring.ts

```diff
@@ -84,26 +84,25 @@ export function withSpring(
               t,
             });
 
-      if (!config.useDuration) {
-        const { isOvershooting, isVelocity, isDisplacement } =
-          isAnimationTerminatingCalculation(animation, config);
-
-        animation.current = newPosition;
-        animation.velocity = newVelocity;
-
-        const springIsNotInMove =
-          isOvershooting || (isVelocity && isDisplacement);
-
-        if (springIsNotInMove) {
-          if (config.stiffness !== 0) {
-            animation.velocity = 0;
-            animation.current = toValue;
-          }
-          // clear lastTimestamp to avoid using stale value by the next spring animation that starts after this one
-          animation.lastTimestamp = 0;
-          return true;
+      animation.current = newPosition;
+      animation.velocity = newVelocity;
+
+      const { isOvershooting, isVelocity, isDisplacement } =
+        isAnimationTerminatingCalculation(animation, config);
+
+      const springIsNotInMove =
+        isOvershooting || (isVelocity && isDisplacement);
+
+      if (!config.useDuration && springIsNotInMove) {
+        if (config.stiffness !== 0) {
```

Comment:
Could you explain this condition? Why we need to reset velocity only in this case?

Reply:
I don't know but we have been doing that for years: https://github.com/software-mansion/react-native-reanimated/pull/2182/files